tests/model/test_item.py

```diff
@@ -136,3 +138,105 @@ def test_resolve_item_parameters():
     parameters.get("param1").value = 10
     assert parameters.get("param1").value != initial.get("param1").value
     assert resolved.pscalar.value == 10
+
+
+RATES_DICT = {
+    "rates": [
+        ["k1sum", 10, {"vary": False, "non-negative": True}],
+        ["k1", 2.0, {"expr": "$b.1 * $rates.k1sum", "non-negative": True, "vary": False}],
+        ["k2", 8.0, {"expr": "$b.2 * $rates.k1sum", "non-negative": True, "vary": False}],
+        ["k3", 0.54321, {"non-negative": True}],
+    ]
+}
+
+B_DICT = {
+    "b": [
+        ["1", 0.2, {"vary": True, "non-negative": True}],
+        ["2", 0.8, {"expr": "1.0 - $b.1", "non-negative": True, "vary": False}],
+    ]
+}
+
+
+def _add_parameters_to_initial(labels, parameters, initial_parameters):
+    for label in labels:
+        add_to_initial(label, parameters, initial_parameters)
+
+
+def _test_add_to_initial(initial_parameters):
```

Comment:
I get that we call `check_parameter_order` multiple times but just defining `expected_order` and using it multiple times should be fine when you also inline the key retrival.
As for the other functions I see no need at all to define functions in a closure vs. just inlining it since they are only called once.

Reply:
Addressed in [fe69696](https://github.com/glotaran/pyglotaran/pull/1500/commits/fe69696ed3be26218c0d80c756c62dbcc3045f81) where the tests are now inlined. 

The expected_order is not always the same, so that was kept, to be explicit about the expected order per test.